File: Minimum-Time-Difference.cpp
class Solution {
public:
    int findMinDifference(vector<string>& timePoints) {
        sort(timePoints.begin(),timePoints.end());
        int mind=(timePoints[timePoints.size()-1][0]-timePoints[0][0])*600+(timePoints[timePoints.size()-1][1]-timePoints[0][1])*60+(timePoints[timePoints.size()-1][3]-timePoints[0][3])*10+(timePoints[timePoints.size()-1][4]-timePoints[0][4]);
        mind=min(mind,1440-mind);
        for(int i=1;i<timePoints.size();i++){
            int d= (timePoints[i][0]-timePoints[i-1][0])*600+(timePoints[i][1]-timePoints[i-1][1])*60+(timePoints[i][3]-timePoints[i-1][3])*10+(timePoints[i][4]-timePoints[i-1][4]);
            d=min(d,1440-d);
            mind=min(mind,d);
        }
        return mind;
    }
};

File: Minimum-Time-Difference.java
class Solution {
    public int findMinDifference(List<String> timePoints) {
        Collections.sort(timePoints);
        int minDiff = calculateDifference(timePoints.get(timePoints.size() - 1), timePoints.get(0));
        minDiff = Math.min(minDiff, 1440 - minDiff);

        for (int i = 1; i < timePoints.size(); i++) {
            int diff = calculateDifference(timePoints.get(i), timePoints.get(i - 1));
            diff = Math.min(diff, 1440 - diff);
            minDiff = Math.min(minDiff, diff);
        }
        
        return minDiff;
    }

    private int calculateDifference(String time1, String time2) {
        int hourDiff = (time1.charAt(0) - time2.charAt(0)) * 600 
                     + (time1.charAt(1) - time2.charAt(1)) * 60;
        int minDiff = (time1.charAt(3) - time2.charAt(3)) * 10 
                     + (time1.charAt(4) - time2.charAt(4));
        return hourDiff + minDiff;
    }
}